src/Avalonia.Controls/Primitives/TextSelectionHandle.cs

```diff
@@ -0,0 +1,210 @@
+﻿using System;
+using Avalonia.Input;
+using Avalonia.Interactivity;
+using Avalonia.Media;
+
+namespace Avalonia.Controls.Primitives
+{
+    public class TextSelectionHandle : TemplatedControl
```

Comment:
Can this class inherit from `Thumb`? Both types seem have a lot of similar code and behavior.